model/cloudery/service.go

```diff
@@ -60,3 +54,37 @@ func (s *ClouderyService) SaveInstance(inst *instance.Instance, cmd *SaveCmd) er
 
 	return nil
 }
+
+func (s *ClouderyService) HasBlockingSubscription(inst *instance.Instance) (bool, error) {
+	client, err := s.getClient(inst)
+	if err != nil {
+		return false, err
+	}
+
+	url := fmt.Sprintf("/api/v1/instances/%s", url.PathEscape(inst.UUID))
+	res, err := client.Get(url)
+	if err != nil {
+		return false, fmt.Errorf("request failed: %w", err)
+	}
+
+	return hasBlockingSubscription(res), nil
+}
+
+func hasBlockingSubscription(clouderyInstance map[string]interface{}) bool {
+	return clouderyInstance["has_blocking_subscription"] == true
+}
+
+func (s *ClouderyService) getClient(inst *instance.Instance) (*manager.APIClient, error) {
+	cfg, ok := s.contexts[inst.ContextName]
+	if !ok {
+		cfg, ok = s.contexts[config.DefaultInstanceContext]
+	}
+
+	if !ok {
+		return nil, fmt.Errorf("%w: tried %q and %q", ErrInvalidContext, inst.ContextName, config.DefaultInstanceContext)
```

Comment:
What does this mean for self-hosted users? Does returning an error for external ties will block instance deletions?

Reply:
I'd say no but that will be up to the Settings app implementation to decide.
However, I've made some manual tests without any cloudery config and no errors were raised.